### ISSUE TITLE
Error / blank page about sbot.conn

### ISSUE THREAD

**Issue by clehner** [soapdog/patchfox#113]
In 2021.6.1, ssb-conn is expected. I'm using ssb-server which doesn't have ssb-conn. Pages are empty and the error in the console: `sbot.conn is undefined`.
![Screenshot from 2021-06-06 23-54-01](https://user-images.githubusercontent.com/95347/120957769-42e53600-c724-11eb-83cf-92643cb3f003.png)

**Comment by soapdog** [comment]
@clehner I ended up pushing a removal of the code that relied on `ssb-conn` before seeing the PR you sent. The code below your changes was part of trying to implement _Rooms 2.0 HTTP Auth_ but it didn't work, and it couldn't work because the flow needs to happen in the backend, not the front-end. In essence, that code was a leftover from an attempt to implement a feature. 

This [commit](https://github.com/soapdog/patchfox/commit/c06c58dd80e4b4a9802a12196515be27f6975c67) pushes both that removal, a version bump, and a prebuilt webextension in the `dist/` folder. It should work for you and fix #113 

Be aware that there is one last piece of code that relies in `ssb-conn` to work and that is joining rooms 2.0.

From `src/core/platforms/ssb/rooms2.js`:

```
  connectAndRemember: (address) => {
    return new Promise((resolve, reject) => { 
      ssb.sbot.conn.connect(address, (err, data) => {
        if (err) {
          reject(err)
        } else {
          ssb.sbot.conn.remember(address)
          resolve({address, data})
        }
      })
    })
  },
```

I will figure out a way to provide a version of that that doesn't rely on ssb-conn. Do you have any tips about how to do it?

Except for that feature, the rest of Patchfox should work for you now. Unless there are other plugins I'm relying on that you don't have, but that is good because I want to make this more agnostic and would gladly patch patchfox to solve these situations.

**Comment by clehner** [comment]
@soapdog I would suggest to call `sbot.gossip.connect` if `ssb-gossip` is present and not `ssb-conn`.
I don't think a separate "remember" step is needed, since although `ssb-gossip` has `gossip.add`, the `gossip.connect` implementation calls it already internally.
I haven't tested this with actual rooms.
```patch
From 021296dd70aa9e1098663ac50d4d42320608b224 Mon Sep 17 00:00:00 2001
From: "Charles E. Lehner" <cel@celehner.com>
Date: Mon, 14 Jun 2021 21:17:49 -0400
Subject: [PATCH] Support ssb-gossip in connectAndRemember

---
 src/core/platforms/ssb/rooms2.js | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/core/platforms/ssb/rooms2.js b/src/core/platforms/ssb/rooms2.js
index fdad5c9..41beb63 100644
--- a/src/core/platforms/ssb/rooms2.js
+++ b/src/core/platforms/ssb/rooms2.js
@@ -17,11 +17,19 @@ const rooms2 = {
   },
   connectAndRemember: (address) => {
     return new Promise((resolve, reject) => { 
-      ssb.sbot.conn.connect(address, (err, data) => {
+      let gossipOrConn = ssb.sbot.gossip || ssb.sbot.conn
+      if (!gossipOrConn) {
+        return reject(new Error('Missing ssb-gossip or ssb-conn'))
+      }
+      gossipOrConn.connect(address, (err, data) => {
         if (err) {
           reject(err)
         } else {
-          ssb.sbot.conn.remember(address)
+          if (ssb.sbot.conn) {
+            ssb.sbot.conn.remember(address)
+          } else {
+            // gossip.connect calls gossip.add internally
+          }
           resolve({address, data})
         }
       })
-- 
2.30.2
```
